### DISCUSSION TITLE
out parameter in while loop

### DISCUSSION THREAD

**Discussion by FaustVX** [dotnet/csharplang#4876] [Language Ideas] [answered]
In this method, the `output` parameter needs to be defined outside the `while` loop.
```csharp
public static T Ask<T>(string prompt, TryConvert<T> converter)
{
    T output;
    do
    {
        Console.Write(prompt + ":");
    } while (!(Console.ReadLine() is string read && converter(read, out output)));
    return output;
}
```
It would be nice to have the `out var` functionality extended to the `while` condition and access the parameter after the loop,
Like that:
```csharp
public static T Ask<T>(string prompt, TryConvert<T> converter)
{
    do
    {
        Console.Write(prompt + ":");
    } while (!(Console.ReadLine() is string read && converter(read, out var output)));
    return output;
}
```

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/blob/main/meetings/2016/LDM-2016-11-30.md#scope-of-while-condition-expression-variables

**Reply by FaustVX** [reply]
Ok, it would be nice, but I understand why it can't be that way.
